source/slang/hlsl.meta.slang

```diff
@@ -3944,67 +4036,100 @@ void abort();
 // Absolute value (HLSL SM 1.0)
 
 __generic<T : __BuiltinIntegerType>
-__target_intrinsic(hlsl)
-__target_intrinsic(glsl)
-__target_intrinsic(cuda, "$P_abs($0)")
-__target_intrinsic(cpp, "$P_abs($0)")
-__target_intrinsic(spirv, "OpExtInst resultType resultId glsl450 fi(FAbs, SAbs) _0")
 [__readNone]
 [require(cpp_cuda_glsl_hlsl_spirv, sm_2_0_GLSL_140)]
-T abs(T x);
-/*{
+T abs(T x)
+{
+    __target_switch
+    {
+    case hlsl: __intrinsic_asm "abs";
+    case glsl: __intrinsic_asm "abs";
+    case cuda: __intrinsic_asm "$P_abs($0)";
+    case cpp: __intrinsic_asm "$P_abs($0)";
+    case spirv: return spirv_asm {
+        result:$$T = OpExtInst glsl450 SAbs $x
```

Comment:
FAbs?

Reply:
My understanding is that FAbs is for floating type.
Because "x" is __BuiltinIntegerType, I don't see why we should want to use FAbs here.
Please let me know if I am missing something.

Similarly when "x" is __BuiltinFloatingPointType, FAbs should be enough.

Reply:
Ah, I didn't see that __generic declaration above and thought this is for all `T`. This works then.